bigquery/storage/managedwriter/managed_stream.go

```diff
@@ -360,11 +366,29 @@ func (ms *ManagedStream) AppendRows(ctx context.Context, data [][]byte, opts ...
 			ProtoDescriptor: pw.newSchema,
 		}
 	}
-	// proceed to call
-	if err := ms.append(pw); err != nil {
-		// pending write is DOA.
-		pw.markDone(NoStreamOffset, err, ms.fc)
-		return nil, err
+	// Call the underlying append.  The stream has it's own retained context and will surface expiry on
+	// it's own, but we also need to respect any deadline for the provided context.
+	errCh := make(chan error)
```

Comment:
nit: Not sure if you import or use it already but this looks like a great usecase for: [errorgroup](https://pkg.go.dev/golang.org/x/sync/errgroup#WithContext)

Reply:
This particular use case is handling a single request on a bidi stream and supporting cancellation, but it's certainly something we may end up with if we end up with write multiplexing.